xc/common/utils/prjxray_routing_import.py

```diff
@@ -865,10 +866,12 @@ def add_synthetic_edges(conn, graph, node_mapping, grid, synth_tiles):
             assert len(option) > 0, (pin, len(option))
 
             if pin['port_type'] == 'input':
-                tile_type = 'SYN-OUTPAD'
+                tile_type = 'SYN-IOPAD'
```

Comment:
This is very wrong.  The tiletype should be SYN-INPAD or SYN-OUTPAD?

Reply:
You're probably right.  Trying now.

Reply:
This code will definitely need to understand the z-stacking too.

Reply:
In the graph pin_name_map, the key it expects is of the form SYN-IOPAD[x].inpad[x] which is why I made this change here from SYN-OUTPUT to SYN-IOPAD.  Does this mean the pin_name_map is being produced incorrectly somewhere else?

Reply:
That actually makes sense, as the routing interface belongs to the tile.  You need to take care to select the right IPIN or OPIN to bind the synth edge too, otherwise things won't line up right.